src/Database/LSMTree/Internal/MergingTree.hs

```diff
@@ -102,6 +103,21 @@ data PreExistingRun m h =
     PreExistingRun        !(Ref (Run m h))
   | PreExistingMergingRun !(Ref (MergingRun MR.LevelMergeType m h))
 
+newCompletedMerge ::
+     (MonadMVar m, PrimMonad m, MonadMask m)
+  => Ref (Run m h)
+  -> m (Ref (MergingTree m h))
+newCompletedMerge run = mkMergingTree . CompletedTreeMerge =<< dupRef run
+
+-- | Create a new 'MergingTree' representing the merge of an ongoing run.
+-- The usage of this function is primarily to facilitate the reloading of an
+-- ongoing merge from a persistent snapshot.
+newOngoingMerge ::
+     (MonadMVar m, PrimMonad m, MonadMask m)
+  => Ref (MergingRun MR.TreeMergeType m h)
+  -> m (Ref (MergingTree m h))
+newOngoingMerge mr = mkMergingTree . OngoingTreeMerge =<< dupRef mr
```

Comment:
Need `INLINE` or `SPECIALISE` pragmas

Reply:
and indeed on every other function in the module.